sqlglot/parser.py

```diff
@@ -945,14 +948,32 @@ def _parse_create(self) -> t.Optional[exp.Expression]:
             this = self._parse_table(schema=True)
             properties = self._parse_properties()
             if self._match(TokenType.ALIAS):
-                expression = self._parse_select(nested=True)
+                expression = self._parse_select(nested=True, parse_subquery_alias=False)
```

Comment:
why did you need to add this?

Reply:
This `_parse_select` calls `_parse_subquery` internally. If `parse_alias=False` isn't passed to `_parse_subquery` it assumes the token following the subquery is an alias. In this `CREATE TABLE` context, we know for sure that that token is not an alias.

For example, consider a command like `CREATE TABLE AS (SELECT a,b) INDEX (a)`. The token in the "INDEX" position will never be an alias for a subquery.

With default `_parse_subquery`:
![image](https://user-images.githubusercontent.com/1831878/212197631-fbec986f-3326-4fb9-b84d-4f4eb493374a.png)

With `parse_alias=False` passed down to `_parse_subquery`:
![image](https://user-images.githubusercontent.com/1831878/212197746-7aa56b7e-ec84-4747-9163-28654d77e59e.png)

The latter is the output we're looking for.

Is there a better way to accomplish this?

Reply:
i didn't understand what case this would be used for, make sure this is a test case

Reply:
Is this an appropriate test case in test_parser.py `test_select`?

```
self.assertIsNone( 
    parse_one("create table a as (select b from c) index").find(exp.TableAlias)
    )
```

Reply:
ok